include/aktualizr-lite/api.h

```diff
@@ -232,6 +244,13 @@ class AkliteClient {
    */
   TufTarget GetCurrent() const;
 
+  /**
+   * Find the Target to rollback to in the event the current target wasn't
```

Comment:
I am not sure that we need to be so specific about using this API function. The rollback target it returns is valid in any other use-cases too. For example, in the case of "just App" update and rollback it should return a valid "rollback" target. I understand that in this case "current" can be used as "rollback" one.
My point is that it may simplify a client code if `GetRollbackTarget` is used in all rollback cases.

Reply:
This is somewhat true, but is there another way this could ever get used given our current user-facing API? 

This function seems so specific to this one use case that I wish we could somehow embed the `if (client.IsRollback(latest) && current.Name() == latest.Name()) {` logic in main.cc into our API. i.e. somehow `GetLatest` would return the rollback Target instead but somehow also let you know that its not really "latest".

Reply:
I think `GetRollbackTarget()` can be used in all other "rollback" cases, specifically when install() or finalizeInstall() returns error. btw, the api client does finalization implicitly, hiding it from a client while liteclient requires its clients do it explicitly.


Regarding `GetLatest` to return either latest or rollback target, I think it really depends on the approach, explicit vs implicit. I would prefer explicit and more mechanistic API, so clients can have more flexibillity in terms of implementation of their client, e.g. maybe they would like to be aware about rollback target and gets some "approval"/confirmation from upper stack to roll back to it.

Reply:
> I think GetRollbackTarget() can be used in all other "rollback" cases, specifically when install() or finalizeInstall() returns error. btw, the api client does finalization implicitly, hiding it from a client while liteclient requires its clients do it explicitly.

But since the existing API is handling that internally - its not actually possible, right? i.e. the only use case in the current api to use this function is this one app failure situation.

> I think it really depends on the approach, explicit vs implicit

I don't want it to be implicit. But this PR's approach is error-prone in that you have to understand it deeply and when/how to use. It feels a bit like the "Ah - you forget to check for NULL" approach. I'm not quite sure how to make it more clean/explicit. Maybe the `CheckInResult` struct could somehow include this additional context?

Reply:
> > I think GetRollbackTarget() can be used in all other "rollback" cases, specifically when install() or finalizeInstall() returns error. btw, the api client does finalization implicitly, hiding it from a client while liteclient requires its clients do it explicitly.
> 
> But since the existing API is handling that internally - its not actually possible, right? i.e. the only use case in the current api to use this function is this one app failure situation.

I don't think so. The `LiteInstall::Install()` does just install, if it fails then it's up to a client to implement rollback in all three rollback cases ("just app" install failure, ostree+app install failure because of ostree rollback, ostree+app install failure because of app creation or start failure after reboot). My point is that in these three rollback cases a client could use `GetRollbackTarget()`.

Reply:
k - could you help show me where else I need to be checking in the example file? If I'm missing logic - then everyone else doing this will be missing logic as well.

Reply:
The logic I mean is missing because we haven't had GetRollbackTarget() so far, and the rollback for two cases was performed implicitly through `} else {
        auto installer = client.CheckAppsInSync();` branch by syncing of the current but not latest target.
        
Now, since we have `GetRollbackTarget()` then we can do rollback explicitly for all three cases.

In this spot

```
} else {
          LOG_ERROR << "Unable to install target: " << ires;
        }
```
We can print logs saying that we are rolling back.

As for the ostree rollback after reboot case we cannot detect it explicitly and print corresponding error message because the finalization is hidden in Init(). If a client  called it directly then it could detect an error and print message about rolling back to the previous version.

The current version of the client example does the rollback for the two cases via syncing apps, and they are not considered as rollback at all, just installation failure.
So, from this perspective we have just one rollback case, if so,  then this PR is fine.

I guess that's fine as long as there are no two failing targets in raw, then the rollback through app syncing will lead to syncing to the failing target. Calling `GetRollbackTarget` and rolling back to the target it returns will do a correct rollback.